llvm/test/CodeGen/AMDGPU/llvm.amdgcn.ballot.i64.wave32.ll

```diff
@@ -3,6 +3,10 @@
 ; RUN: llc -march=amdgcn -global-isel=0 -mcpu=gfx1100 -amdgpu-enable-delay-alu=0 < %s | FileCheck %s --check-prefixes=CHECK,DAGISEL
 ; RUN: llc -march=amdgcn -global-isel -mcpu=gfx1010 < %s | FileCheck %s --check-prefixes=CHECK,GISEL
 ; RUN: llc -march=amdgcn -global-isel -mcpu=gfx1100 -amdgpu-enable-delay-alu=0 < %s | FileCheck %s --check-prefixes=CHECK,GISEL
+; RUN: opt -mtriple=amdgcn-- -mcpu=gfx1010 -mattr=+wavefrontsize32,-wavefrontsize64 -passes=instcombine -o - < %s | llc -march=amdgcn -global-isel=0 -mcpu=gfx1010 - | FileCheck %s --check-prefixes=CHECK-OPT,DAGISEL-OPT
```

Comment:
These tests should go in test/Transforms/InstCombine/AMDGPU, not complicate the codegen tests

Reply:
I see that tests in _test/Transforms/InstCombine/AMDGPU_ expectedly run only _opt_ but I would like to test the behavior of the entire compilation in -O0 and higher modes. In particular I would like to check if the _branch on ballot.i64!=0_ selection works correctly in wave32 mode. So this is more a codegen test. We have _llvm/test/Transforms/InstCombine/AMDGPU/amdgcn-intrinsics.ll_ that checks ballot.

Reply:
They're really 2 separately testable pieces, and this doesn't show you the entire compilation output (especially the -O0 bit). We don't really have anywhere that does end to end pipeline tests, but clang has a few.